### PR TITLE
feat(connector): optionally limit kafka input throughput bytes

### PR DESCRIPTION
**Pull request by lmatz**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Add a configurable parameter to limit the source throughput in terms of bytes.

This parameter may or may not(so not go into doc and release notes this time) be intended to be used by public users at the moment.

It is convenient for people to test resource utilization/performance for a fixed input throughput.(https://redpanda.com/blog/redpanda-vs-kafka-performance-benchmark#:~:text=3.2%20under%20workloads%20(-,up%20to%201GB/sec,-)%20that%20are%20common)
Otherwise, people have to always generate events by some source generators on the fly to Kafka/Redpanda at exactly the throughput you want, which may not be achievable and depends on many other factors.
Instead, we can pre-generate a lot and then limit the source throughput stably.

Also, it may be a workaround solution for #5214.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] ~I have added necessary unit tests and integration tests~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Comment by lmatz** [comment]
```
create source kafkas(v1 bigint) with (
  connector = 'kafka',
  kafka.brokers = '127.0.0.1:9092',
  kafka.topic = 'sink_target',
  scan.startup.mode = 'earliest',
  kafka.bytes.per.second = '20000'
) row format json;
```

```
parallelism: 4
```
SCR-20221226-oux

File: src/connector/src/source/kafka/mod.rs [hunk 1]
```diff
@@ -29,6 +29,9 @@ pub const KAFKA_CONNECTOR: &str = "kafka";
 
 #[derive(Clone, Debug, Deserialize)]
 pub struct KafkaProperties {
+    #[serde(rename = "bytes.per.second", alias = "kafka.bytes.per.second")]
+    pub bytes_per_second: Option<String>,
```
Comment (tabVersion) [inline src/connector/src/source/kafka/mod.rs:33]: per executor or per partition or sth?
please add some comments here

**Review by tabVersion — APPROVED** [review]
rest LGTM

**Comment by tabVersion** [comment]
It can work as a temp fix for fine-tuning the performance but not a user-facing option.

**Pull request merged**